packages/components/src/Button.tsx

```diff
@@ -31,6 +36,7 @@ interface BaseButtonProps extends React.ComponentPropsWithRef<'button'> {
   icon?: IconDefinition | JSX.Element;
   active?: boolean;
   'data-testid'?: string;
+  dataAttributes?: DataAttribute;
```

Comment:
As mentioned in my other comment, I don't think we should have a `dataAttributes` prop on button. We should instead look at refactoring the `AdvancedFilterCreator` data attribute usage so that we don't need to use data attributes there, then we don't need to put data attributes on the button itself.